### PR TITLE
Python 3.11.2 Support

### PR DESCRIPTION
**Pull request by epps**
# Description

This PR introduces minor changes required to support the latest version of Python (i.e. 3.11.2).

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Language Support

# Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: client/setup.cfg [hunk 1]
```diff
@@ -26,15 +26,15 @@ install_requires =
     protobuf>=3.20.0
     typeguard<3.0.0
     grpcio>=1.47.0
-    numpy==1.21.6
+    numpy
```
Comment (simba-git) [inline client/setup.cfg:29]: Any argument to set this to a >=

File: client/setup.cfg [hunk 2]
```diff
@@ -26,15 +26,15 @@ install_requires =
...
     pandas==1.3.5
     pandasql>=0.7.3
     typing_extensions>=4.1.1
     dataclasses==0.6
     flask==2.2.1
     Flask-Cors==3.0.10
     validators>=0.20.0
-    dill==0.3.5.1
+    dill
```
Comment (simba-git) [inline client/setup.cfg:37]: Same

File: client/src/featureform/register.py [hunk 3]
```diff
@@ -736,15 +736,8 @@ def __init__(self, fn, registrar, provider, decorator_register_resources_method,
...
-        # Binds the register_resources and name_variant methods to the subscriptable_fn
-        # using the descriptor protocol. This allows us to call the methods on the
-        # subscriptable_fn instance. **NOTE** the MethodType could also be used to bind
-        # the methods to the subscriptable_fn instance; however, the descriptor protocol
-        # produces the same result while also being compatible with `classmethod`
-        # and `staticmethod`.
-        self.register_resources = decorator_register_resources_method.__get__(self)
-        self.name_variant = decorator_name_variant_method.__get__(self)
-        pass
+        self.register_resources = decorator_register_resources_method
```
Comment (simba-git) [inline client/src/featureform/register.py:739]: Ironically this is originally how I thought we should do this, and the whole __get__ thing confused the hell out of me. I just assume you can use fns and methods however you want in python.

**Review by simba-git — APPROVED** [review]

File: client/src/featureform/register.py (hunk 3 above)
Comment (epps) [reply client/src/featureform/register.py:739]: I probably overcomplicated this based on a misguided bias from JS. If this were JS, I would never expect the value of `self` to be correct unless I bound it. When I searched for how to do this in Python, I happened upon the approach, which was touted as better than just a "raw" assignment or using `types.MethodType`. Lesson learned.

File: client/setup.cfg (hunk 1 above)
Comment (epps) [reply client/setup.cfg:29]: I'm not sure I'm following your comment here. Are you saying you'd prefer some `>=` specification?

**Review by simba-git — APPROVED** [review]

**Review by aolfat — APPROVED** [review]

**Pull request merged**